gradle/license.gradle

```diff
@@ -0,0 +1,16 @@
+def licenseHeaderFile = rootProject.file('doc/licenses/HEADER.txt')
+
+allprojects {
+    apply plugin: 'com.github.hierynomus.license'
```

Comment:
Do we use that plugin to handle any other type of files than `.java` and `.groovy` (`.kt`?)? If not, maybe it would be good to narrow the scope to only submodules with code? It can be achieved the same way as with the coverage: `plugins.withId("java") { ... }`.

Reply:
Yes also any configuration or extra files. We can limit to Java projects, but I would rather check that all our files comply. Apparently we were missing quite some in total and the more we can catch, the better. WDYT?

Reply:
If it's also for non source code files I'm ok with `allprojects` unconditionally.